dev/com.ibm.ws.kernel.feature.core/src/com/ibm/ws/kernel/feature/internal/subsystem/FeatureRepository.java

```diff
@@ -442,8 +460,14 @@ static ImmutableAttributes loadFeatureAttributes(DataInputStream in) throws IOEx
             processTypes.add(valueOf(in.readUTF(), ProcessType.SERVER));
         }
         ActivationType activationType = valueOf(in.readUTF(), ActivationType.SEQUENTIAL);
-        return new ImmutableAttributes(repositoryType, symbolicName, shortName, featureVersion, visibility, appRestart, version, featureFile, lastModified, fileSize, isAutoFeature,
-                                       hasApiServices, hasApiPackages, hasSpiPackages, isSingleton, processTypes, activationType);
+        short altNamesCount = in.readShort();
+        Set<String> altNames = new HashSet<>();
```

Comment:
might as well seed the size here for the collection.  If this is an ArrayList that would make it pretty small.  Should consider using Collections.empty for when size == 0